packages/task/src/browser/task-service.ts

```diff
@@ -94,6 +97,8 @@ export class TaskService implements TaskConfigurationClient {
         terminateSignal: Deferred<string | undefined>,
         isBackgroundTaskEnded: Deferred<boolean | undefined>
     }>();
+    // URIs of the opened documents
+    protected openedDocuments: Set<string> = new Set();
```

Comment:
please don't introduce caches, caches are evil because can cause race condition, memory leaks and so on

they are necessary sometimes for performance, like for replicate data structures between 2 processes

but even in this case they should be properly encapsulated and respective service should be used.

In our case `MonacoWorkspace` know about opened document and should be asked about the current state.